### ISSUE TITLE
Support Parallel Test Execution

### ISSUE THREAD

**Issue by plioi** [fixie/fixie#46]
Test execution is currently sequential.

To enable parallel tests, we likely want to address the _outermost_ loop:

```
foreach (var testClass in convention.Classes.Filter(candidateTypes))
```

Originally I wrote here that the offenders are the foreach within CreateInstancePerCase and ExecuteCases, but I think we don't want to parallelize _within_ a given test class.  There are too many complications parallelizing intra-test-class.

Should parallelization be enabled for the whole run ("from the outside") or explicitly within convention classes?

**Comment by circa1741** [comment]
FYI only. But it would be nice to have it in Fixie.

The upcoming NUnit 3.0 has the ParallelizableAttribute. The specification says that it should be able to run in the assembly, class or method level. I believe it currently can run in the class level.

https://github.com/nunit/dev/wiki/Framework-Parallel-Test-Execution

**Comment by Rookian** [comment]
@plioi Can you please provide a new status on this issue? It would be nice to be able to run unit tests in parallel.

**Comment by plioi** [comment]
It's still quite on the back-burner. It would likely be disruptive enough to warrant a 3.0 release.

**Comment by execution** [comment]
Any update for this issue?

**Comment by plioi** [comment]
Closing this issue due to age, and the fact that the newly released version 3 API is dramatically different from this discussion.

Version 3.0.0 doesn't have parallel test running support either, and it is very unlikely to gain *in-process* parallelism ever. I'd argue it's more harmful than good to the stability of the runner.

A more sound possibility would be *process*-level parallelism, and that would be a matter of improving the way the "dotnet fixie" command farms out work to the test project processes it kicks off today. You could get a poor imitation of that parallelism today by kicking off "dotnet fixie" multiple times at once with completementary --tests wildcards.
